improver/developer_tools/metadata_interpreter.py

```diff
@@ -0,0 +1,563 @@
+                            f"Title {attrs['title']} is inconsistent with model ID "
+                            f"attribute {attrs[self.model_id_attr]}"
+                        )
+
+                try:
+                    self.model = MODEL_NAMES[attrs[self.model_id_attr]]
+                except KeyError:
+                    self.errors.append(
+                        f"Attribute {attrs[self.model_id_attr]} is not a valid single "
+                        "model.  If valid for blend, then title attribute is missing "
+                        f"expected substring {BLEND_TITLE_SUBSTR}."
+                    )
+
+    def check_attributes(self, attrs):
+        """Checks for unexpected attributes, then interprets values for model
+        information and checks for self-consistency"""
+        try:
+            permitted_attributes = COMP_ATTRS + DIAG_ATTRS[self.diagnostic]
+        except KeyError:
+            permitted_attributes = COMP_ATTRS.copy()
+
+        if any([attr in NONCOMP_ATTRS for attr in attrs]):
+            self.errors.append(
+                f"Attributes {attrs.keys()} include one or more forbidden "
+                f"values {NONCOMP_ATTRS}"
+            )
+        elif any([attr not in permitted_attributes for attr in attrs]):
+            self.warnings.append(
+                f"{attrs.keys()} include unexpected attributes. Please check the "
+                "standard to ensure this is valid."
+            )
+
+        if self.diagnostic in DIAG_ATTRS:
+            required = DIAG_ATTRS[self.diagnostic]
+            if any([req not in attrs for req in required]):
+                self.errors.append(
+                    f"Attributes {attrs.keys()} missing one or more required "
+                    f"values {required}"
+                )
+
+        if self.field_type != self.ANCIL:
+            if not all([attr in attrs for attr in MANDATORY_ATTRIBUTES]):
+                self.errors.append(
+                    f"Attributes {attrs.keys()} missing one or more mandatory "
+                    f"values {MANDATORY_ATTRIBUTES}"
+                )
+
+            try:
+                self.post_processed = (
+                    True
+                    if PP_TITLE_SUBSTR in attrs["title"]
+                    or BLEND_TITLE_SUBSTR in attrs["title"]
+                    else False
+                )
+            except KeyError:
+                self.errors.append("Cube is missing mandatory title attribute")
+            else:
+                self._check_blend_and_model_attributes(attrs)
+
+    def _check_coords_present(self, coords, expected_coords):
+        """Check whether all expected coordinates are present"""
+        found_coords = [coord for coord in coords if coord in expected_coords]
+        if not set(found_coords) == set(expected_coords):
+            self.errors.append(
+                f"Missing one or more coordinates: found {found_coords}, "
+                f"expected {expected_coords}"
```

Comment:
This could be simplified to just list the missing coords `set(expected_coords) - set(found_coords)`

Reply:
Yes, it could, but in this instance I don't want it to.  Because for blended data, we should have a completely different set of time coordinates than unblended data.  So I want the `expected_coords` message in this case to give the full list.

Reply:
Ok